library/x509write_crt.c

```diff
@@ -293,6 +293,44 @@ int mbedtls_x509write_crt_set_key_usage( mbedtls_x509write_cert *ctx,
     return( 0 );
 }
 
+int mbedtls_x509write_crt_set_ext_key_usage( mbedtls_x509write_cert *ctx,
+                                             const mbedtls_asn1_sequence *exts )
+{
+    unsigned char buf[256];
+    unsigned char *c = buf + sizeof(buf);
+    int ret;
+    size_t len = 0;
+    const mbedtls_asn1_sequence *last_ext = 0, *ext;
+
+    /* We need at least one extension: SEQUENCE SIZE (1..MAX) OF KeyPurposeId */
+    if( exts == NULL )
+        return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
+
+    /* Iterate over exts backwards, so we write them out in the requested order */
+    while( last_ext != exts )
+    {
+        for( ext = exts; ext->next != last_ext; ext = ext->next ) {}
```

Comment:
What is the expected behaviour out of this line? Is that a control flow mechanism to iterate unti `ext->next == ext_last` and then execute the code under the `while( last_ext != exts )` just once? Or are we missing something?

Reply:
IIUC it's running through the linked list to set `ext` to the entry just before the one that was handled in the previous loop iteration (i.e., find the one whose `next` pointer points to `last_ext`). This seems correct to me.